pycheribuild/projects/cross/cheribsd.py

```diff
@@ -1678,6 +1678,8 @@ def default_kernel_config(self, platform: "Optional[ConfigPlatform]" = None, **f
 
     def extra_kernel_configs(self) -> "list[CheriBSDConfig]":
         # Everything that is not the default kernconf
+        if self.kernel_config != self.default_kernel_config():
+            return []
```

Comment:
How about the following?
```suggestion
        option = inspect.getattr_static(self, "kernel_config")
        assert isinstance(option, ConfigOptionBase)
        if not option.is_default_value:
            return []
```

Reply:
Uhm this seems to break a bunch of tests. I need to understand exactly what is going on there.

Reply:
Upon further analysis, this seems to be tripping on an assertion because `option.is_default_value` can only be used after the `ConfigOptionBase._cached` has been loaded. The check in ConfigOptionBase may be conflating the _cached value with having loaded the value, this seems to be the case when no default is available, e.g. when has_default_buildkernel_kernel_config is set.

The test infrastructure seems to be triggering `ConfigOptionBase.load_option` multiple times, leading to the _is_default_value being retained erroneously. I'm not entirely sure how this is happening, it might be that the cached value is None and we load the value again. This however does not explain why it was marked as default before. This seems like a racy thing that happens only when multiple tests run.

Reply:
Hmm the logic to reset in the unit tests is somewhat hacky so it does sound like something is going wrong there...